server/src/main/java/org/opensearch/rest/action/admin/AdminAPIsUtility.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin;
+
+import org.opensearch.action.admin.cluster.node.tasks.create.CreateTaskRequest;
+import org.opensearch.action.admin.cluster.node.tasks.create.CreateTaskResponse;
+import org.opensearch.action.admin.cluster.node.tasks.unregister.UnregisterTaskRequest;
+import org.opensearch.action.admin.cluster.node.tasks.unregister.UnregisterTaskResponse;
+import org.opensearch.action.support.TimeoutTaskCancellationUtility;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.tasks.CancellableTask;
+import org.opensearch.tasks.Task;
+
+public class AdminAPIsUtility {
+
+    public CancellableTask createAndCancelCancellableTask(NodeClient client, TimeValue timeout) {
```

Comment:
@aasom143 : I am thinking if every admin rest action had its own transport class, then we dont need to explicitly create right?

Reply:
No Shweta, Not every rest action has it's transport class. For RestShardsAction action it directly call two other transport actions at rest layer so that why we have to create a new task at rest layer.

Reply:
Yes we can add it right?

Reply:
this way it would remain consistent and you wont need to add creatsTask, unregisterTask any of these support.